src/utils/decimals.ts

```diff
@@ -0,0 +1,64 @@
+import { Address, Chain, PublicClient, Transport, zeroAddress } from 'viem';
+import { fetchDecimals } from './erc20';
+
+export async function getNativeTokenDecimals<TChain extends Chain | undefined>({
+  publicClient,
+  nativeTokenAddress,
+}: {
+  publicClient: PublicClient<Transport, TChain>;
+  nativeTokenAddress: Address;
+}) {
+  if (!nativeTokenAddress || nativeTokenAddress === zeroAddress) {
+    return 18;
+  }
+
+  try {
+    return await fetchDecimals({ address: nativeTokenAddress, publicClient });
+  } catch {
+    return 0;
+  }
+}
+
+export function scaleToNativeTokenDecimals({
```

Comment:
Version [4.0.2-beta.0](https://www.npmjs.com/package/@arbitrum/sdk/v/4.0.2-beta.0) of Arbitrum SDK has these utils but for Ethers, let's just create bigint wrappers